test/middleware/session.middleware.test.ts

```diff
@@ -52,38 +50,38 @@ describe("Session Middleware", () => {
 
             await SessionMiddleware(config, sessionStore)(request, Substitute.for<express.Response>(), nextFunction);
 
-            expect(request.session.isNothing()).to.eq(true);
+            expect(request.session).to.eq(undefined);
             sessionStore.didNotReceive().load(Arg.any());
         });
     });
 
     describe("when cookie is present", () => {
-        const session: VerifiedSession = getValidSessionObject(config);
-        const cookie: Cookie = Cookie.representationOf(session, config.cookieSecret);
-        const request = { cookies: { [config.cookieName]: cookie.value } } as express.Request;
+        const sessionData = createSessionData(config.cookieSecret)
+        const session: Session = new Session(sessionData);
+        const request = { cookies: { [config.cookieName]: "" + session.get(SessionKey.Id) + session.get(SessionKey.ClientSig) } } as express.Request;
         const cookieArg = () => {
-            return Arg.is(_ => _.value === cookie.value);
+            return Arg.is(_ => _.value === "" + session.get(SessionKey.Id) + session.get(SessionKey.ClientSig));
         };
 
         it("should load a session and insert the session object in the request", async () => {
             const sessionStore = Substitute.for<SessionStore>();
-            sessionStore.load(cookieArg()).returns(wrapEither(Either.of(session.data)));
+            sessionStore.load(cookieArg()).returns(Promise.resolve(sessionData));
 
             await SessionMiddleware(config, sessionStore)(request, Substitute.for<express.Response>(), nextFunction);
 
-            expect(request.session.isJust()).to.equal(true);
-            expect(request.session.extract().data).to.be.deep.equal(session.data);
+            // @ts-ignore
```

Comment:
do we need ts-ignore here?

Reply:
Already removed but historically it was supported to give access to internal sessions data. In the end I decided to restore public access to session data.